docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md

```diff
@@ -1,13 +1,69 @@
+
+## API Gateway
+
+The eShopOnContainers application has multiple front end clients and multiple different backend services. There is no one-to-one correspondence between the client applications and the microservices that support them. In such a scenario, there may be a great deal of complexity when writing client software to interface with the various backend services in a secure manner. Each client would need to address this complexity on its own, resulting in duplication and many places in which to make updates as services change or new policies are implemented.
+
+Azure API Management (APIM) helps organizations publish APIs in a consistent, manageable fashion. APIM consists of three components: the API Gateway, and administration portal (the Azure portal), and a developer portal.
+
+The API Gateway accepts API calls and routes them to the appropriate backend API. It can also provide additional services like verification of API keys or JWT tokens and API transformation on the fly without code modifications (for instance, to accommodate clients expecting an older interface).
+
+The Azure portal is where you define the API schema and package different APIs into products. You also configure user access, view reports, and configure policies for quotas or transformations.
+
+The developer portal serves as the main resource for developers. It provides developers with API documentation, an interactive test console, and reports on their own usage. Developers also use the portal to create and manage their own accounts, including subscription and API key support.
+
+Using APIM, the eShopOnContainers application exposes several different products of services, each providing a backend for a particular front end client.
+
+## Data
+
+The various backend services used by eShopOnContainers have different storage requirements. Several microservices use SQL Server databases. The Basket microservice leverages a Redis cache for its persistence. The Locations microservice expects a MongoDB API for its data. Azure supports each of these data formats.
+
+For SQL Server database support, Azure has products for everything from single databases up to highly scalable SQL Database elastic pools. Individual microservices can be configured to communicate with their own individual SQL Server databases quickly and easily. These can be scaled as needed to support each separate microservice according to its needs.
+
+The eShopOnContainers application stores the user's current shopping basket between requests. This is managed by the Basket microservice which stores the data in a Redis cache. In development, this cache can be deployed in a container, while in production in can leverage Azure Cache for Redis. Azure Cache for Redis is a fully managed service offering high performance and reliability without the need to deploy and manage your own Redis instances or containers.
```

Comment:
```suggestion
The eShopOnContainers application stores the user's current shopping basket between requests. This is managed by the Basket microservice that stores the data in a Redis cache. In development, this cache can be deployed in a container, while in production it can utilize Azure Cache for Redis. Azure Cache for Redis is a fully managed service offering high performance and reliability without the need to deploy and manage Redis instances or containers on your own.
```